google_guest_agent/windows_accounts_test.go

```diff
@@ -218,25 +218,60 @@ func TestCompareAccounts(t *testing.T) {
 
 func TestRemoveExpiredKeys(t *testing.T) {
 	var tests = []struct {
-		key   string
-		valid bool
+		key      string
+		valid    bool
+		userKeys string
 	}{
-		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0000"}`, true},
-		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0700"}`, true},
-		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0700", "futureField": "UNUSED_FIELDS_IGNORED"}`, true},
-		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2018-11-08T19:30:46+0000"}`, false},
-		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2018-11-08T19:30:46+0700"}`, false},
-		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"INVALID_TIMESTAMP"}`, false},
-		{`user:ssh-rsa [KEY] google-ssh`, false},
-		{`user:ssh-rsa [KEY] user`, true},
-		{`user:ssh-rsa [KEY]`, true},
-		{},
+		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0000"}`,
+			true,
+			`ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0000"}`,
+		},
+		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0700"}`,
+			true,
+			`ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0700"}`,
+		},
+		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0700", "futureField": "UNUSED_FIELDS_IGNORED"}`,
+			true,
+			`ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2028-11-08T19:30:47+0700", "futureField": "UNUSED_FIELDS_IGNORED"}`,
+		},
+		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2018-11-08T19:30:46+0000"}`,
+			false,
+			``,
+		},
+		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"2018-11-08T19:30:46+0700"}`,
+			false,
+			``,
+		},
+		{`user:ssh-rsa [KEY] google-ssh {"userName":"user@email.com", "expireOn":"INVALID_TIMESTAMP"}`,
+			false,
+			``,
+		},
+		{`user:ssh-rsa [KEY] google-ssh`,
+			false,
+			``,
+		},
+		{`user:ssh-rsa [KEY] user`,
+			true,
+			`ssh-rsa [KEY] user`,
+		},
+		{`user:ssh-rsa [KEY]`,
+			true,
+			`ssh-rsa [KEY]`,
+		},
+		{`malformed-ssh-keys [KEY] google-ssh`,
+			false,
+			``,
+		},
+		{`:malformed-ssh-keys [KEY] google-ssh`,
+			false,
+			``,
+		},
 	}
 
 	for _, tt := range tests {
 		ret := removeExpiredKeys([]string{tt.key})
 		if tt.valid {
-			if len(ret) == 0 || ret[0] != tt.key {
+			if len(ret) == 0 || ret[`user`][0] != tt.userKeys {
```

Comment:
this testing paradigm made sense when removeExpiredKeys only pruned, and we were  testing one key at a time, either it is pruned or not
now we still want to know whether keys have been removed

try replacing this entire section starting with `if tt.valid {` with:
```
if userKeys, ok := ret['user']; !ok || len(ret) != tt.expectedValid {
  t.Errorf("expected %d valid keys from getUserKeys", tt.expectedValid)
}
```